chapters/data-analysis.tex

```diff
@@ -1,292 +1,488 @@
-they will become cumbersome to update after revisions.
-If you are writing your code in Git/GitHub,
-you should output results into that directory,
-not the directory where data is stored (i.e. Dropbox).
-Results will differ across different branches
-and will constantly be re-writing each other:
-you should have each result stored with the code that created it.
 
-%------------------------------------------------
+% Exploratory and final data analysis -----------------------------------------
+The analysis stage usually starts with a process we call exploratory data analysis.
+This is when you are trying different things and looking for patterns in your data. 
+It progresses into final analysis when your team starts to decide what are the main results, those that will make it into the research output.
+The way you deal with code and outputs for exploratory and final analysis is different, and this section will discuss how.
+% Organizing scripts ---------------------------------------------------------
+During exploratory data analysis, you will be tempted to write lots of analysis into one big, impressive, start-to-finish script. 
+Though it's fine to write such a script during a long analysis meeting, this practice is error-prone.  
+It subtly encourages poor practices such as not clearing the workspace and not loading fresh data. 
+It's important to take the time to organize scripts in a clean manner and to avoid mistakes.
+
+A well-organized analysis script starts with a completely fresh workspace and explicitly loads data before analyzing it.
+This encourages data manipulation to be done earlier in the workflow (that is, during construction).
+It also and prevents you from accidentally writing pieces of analysis code that depend on one another, leading to the too-familiar ``run this part, then that part, then this part'' process.
+Each script should run completely independently of all other code.
+You can go as far as coding every output in a separate script.
+There is nothing wrong with code files being short and simple -- as long as they directly correspond to specific pieces of analysis.
+
+Analysis files should be as simple as possible, so you can focus on the econometrics.
+All research decisions should be made very explicit in the code.
+This includes clustering, sampling, and control variables, to name a few. 
+If you have multiple analysis data sets, each of them should have a descriptive name about its sample and unit of observation.
+As your team comes to a decision about model specification, you can create globals or objects in the master script to use across scripts.
+This is a good way to make sure specifications are consistent throughout the analysis. It's also very dynamic, making it easy to update all scripts if needed.
+It is completely acceptable to have folders for each task, and compartmentalize each analysis as much as needed.
+It is always better to have more code files open than to keep scrolling inside a given file.
```

Comment:
"_It is always better **to have** to have more code files open than to keep scrolling inside a given file._" or "_It is always better **to be required** to have more code files open than to keep scrolling inside a given file._" Just having more code files open is not the trade-off we are talking about here. Does that make sense?

Reply:
Not sure I get it. Let's talk about in the meeting.

Reply:
I actually think we should scrape this sentence altogether. You shouldn't need multiple files open at once if they are independent of each other.
```suggestion
```

Reply:
done in 13c1f85